queries/ecma/highlights.scm

```diff
@@ -30,24 +30,71 @@
  (#lua-match? @constant "^_*[A-Z][A-Z%d_]*$"))
 
 ((identifier) @variable.builtin
- (#vim-match? @variable.builtin "^(arguments|module|console|window|document)$"))
+ (#any-of? @variable.builtin
+  "arguments"
+  "module"
+  "console"
+  "window"
+  "document"
+  "Object"
+  "Function"
+  "Boolean"
+  "Symbol"
+  "Number"
+  "Math"
+  "Date"
+  "String"
+  "RegExp"
+  "Map"
+  "Set"
+  "WeakMap"
+  "WeakSet"
+  "Promise"
+  "Intl"
+  "Array"
+  "Int8Array"
+  "Uint8Array"
+  "Uint8ClampedArray"
+  "Int16Array"
+  "Uint16Array"
+  "Int32Array"
+  "Uint32Array"
+  "Float32Array"
+  "Float64Array"
+  "ArrayBuffer"
+  "DataView"
+  "Error"
+  "EvalError"
+  "InternalError"
+  "RangeError"
+  "ReferenceError"
+  "SyntaxError"
+  "TypeError"
+  "URIError"))
```

Comment:
These should be `@type.builtin`.

Reply:
this should be separated into `@variable.builtin` for arguments, module, windows, console, document and `type.builtin` for everything else i think

Reply:
Hmm, on second look, `Promise` and `Intl` aren't exactly types though?
Probably more separation is necessary

Reply:
> Hmm, on second look, `Promise` and `Intl` aren't exactly types though?

They are classes.

Reply:
Intl is almost never used as types in typescript im pretty sure. Promise im okay with it as type

Reply:
For consistencies, I agree.

In general usage, probably not. I will just override locally, that's probably better

Reply:
Maybe `@namespace.builtin` for `Intl`?

Reply:
Definitely more correct than `@type.builtin` to me